### PR TITLE
ANDROAPP-5563-Implement-InputMatrix-and-InputSequential

### PR DESCRIPTION
**Pull request by xavimolloy**
## Description
Please include a summary of the change and include the related jira issue if it exists.

[ jira issue ](https://dhis2.atlassian.net/browse/ANDROAPP-

## Solution description
If this PR is a fix include a brief description on how the issue is solved.
## Covered unit test cases
Describe the tests that you ran to verify your changes.
## Where did you test this issue?
- [ ] Smartphone Emulator
- [ ] Tablet Emulator
- [ ] Smartphone
- [ ] Tablet
## Which Android versions did you test this issue?
- [ ] 4.4
- [ ] 5.X - 6.X
- [ ] 7.X
- [ ] 8.X
- [ ] 9.X - 10.X
- [ ] 11.X - 13.X
- [ ] Other
## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code

### CODE REVIEW

File: form/src/main/java/org/dhis2/form/ui/provider/inputfield/FieldProvider.kt [hunk 1]
```diff
@@ -623,6 +645,132 @@ private fun ProvideEmail(
     )
 }
 
+@Composable
+private fun ProvideMatrixInput(
```
Comment (andresmr) [inline form/src/main/java/org/dhis2/form/ui/provider/inputfield/FieldProvider.kt:649]: Please, could you extract this provides to an external class to reduce FieldProvider size?.

**Review by andresmr — CHANGES_REQUESTED** [review]

File: form/src/main/java/org/dhis2/form/ui/provider/inputfield/FieldProvider.kt (hunk 1 above)
Comment (xavimolloy) [reply form/src/main/java/org/dhis2/form/ui/provider/inputfield/FieldProvider.kt:649]: Of course, I'll update it

**Review by andresmr — APPROVED** [review]

File: form/src/main/java/org/dhis2/form/ui/provider/inputfield/MatrixInputProvider.kt [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+) {
+    val inputCardDataList = remember {
+        mutableListOf<IconCardData>()
+    }
+    var matrixSelectedItem by remember(fieldUiModel) { mutableStateOf<IconCardData?>(null) }
+
+    fieldUiModel.optionSetConfiguration?.optionsToDisplay()?.forEach() { option ->
+        val color =
+            ObjectStyleUtils.getColorResource(context, option.style().color(), R.color.colorPrimary)
+        option.style().icon()?.let {
```
Comment (mmmateos) [inline form/src/main/java/org/dhis2/form/ui/provider/inputfield/MatrixInputProvider.kt:36]: This let block is not been used?

**Review by mmmateos — APPROVED** [review]

File: form/src/main/java/org/dhis2/form/ui/provider/inputfield/MatrixInputProvider.kt (hunk 2 above)
Comment (xavimolloy) [reply form/src/main/java/org/dhis2/form/ui/provider/inputfield/MatrixInputProvider.kt:36]: you are correct, I will remove it

**Pull request merged**